### DISCUSSION TITLE
How to add DataStream to Container

### DISCUSSION THREAD

**Discussion by RaubCamaioni** [PyAV-Org/PyAV#1636] [1. Help] [answered]
Working with mpegts videos that container KLV data. ffprove output:
Input #0, mpegts, from 'day_flight.mpg':
--Duration: 00:03:14.88, start: 10.000000, bitrate: 4187 kb/s
--Program 1 
--Stream #0:0[0x1e1]: Video: h264 (Main) ([27][0][0][0] / 0x001B), yuv420p(progressive), 1280x720, 60 fps, 60 tbr, 90k tbn, 180k tbc
--Stream #0:1[0x1f1]: Data: klv (KLVA / 0x41564C4B)
  
Download video from: https://samples.ffmpeg.org/MPEG2/mpegts-klv/

I am unable to determine the syntax for adding a data stream to a container. 
The default example syntax for adding a video stream based on templates does not work for data streams. 

Here is the example code I am using to test different codec_name and parameters. 

python strip_data.py -i day_flight.mpg -o klv.data
``` python
# strip_data.py
from contextlib import closing, ExitStack
import av


def extract_data_stream(src_file, dst_file):
    with ExitStack() as stack:
        src_c = stack.enter_context(closing(av.open(src_file, mode="r")))
        src_s = src_c.streams.data[0]

        # I have tried multiple formats
        dst_c = stack.enter_context(closing(av.open(dst_file, format="data", mode="w")))
        # How to add "data" stream
        dst_s = dst_c.add_stream(codec_name="?", template=src_s)

        for src_p in src_c.demux(src_s):
            for dst_p in dst_s.encode(src_p):
                dst_c.mux(dst_p)


if __name__ == "__main__":
    import argparse

    parser = argparse.ArgumentParser()
    parser.add_argument("-i", type=str, required=True)
    parser.add_argument("-o", type=str, required=True)
    args = parser.parse_args()

    extract_data_stream(args.i, args.o)
```

**Comment by WyattBlue** [comment, marked as answer]
It's not possible to add datastreams in the latest stable version of PyAV right now
